### ISSUE TITLE
How to integrate XMRig version 6.21.3, I find XMRig 6.21.3 more stable 

### ISSUE THREAD

**Issue by dlsofree** [UnamSanctam/SilentCryptoMiner#2361]
How to integrate XMRig version 6.21.3, I find XMRig 6.21.3 more stable

**Comment by UnamSanctam** [comment]
I can do so for the next version, just haven't figured out what other things I should add/change yet.

**Comment by Alcinzal** [comment]
I have a few suggestions that I have written down. Some are pointless really.
* On the GUI, the top half of some elements dont work, like the exit button (very minuscule change).
* Making the GUI for CPU and GPU look the same, currently there are a few differences.
* Have the ability to inject into itself, instead of having to inject to another process.
* Some sort of dynamic mining feature, might be difficult. Like if you set it to dynamic max 50%, then it would mine at 50% if no other processes are using the CPU/GPU, however if a process is using 10%, the miner changes to 40%, and if processes are using 25%, the miner will mine at 25%, and so on. And if processes are using 60%, then the miner will mine at 0%.
* I have written down "Uninstall not working properly", but I don't really remember what was the issue here. I might have resolved this in another issue post.
* I also wrote down "If I load UnamSanctams full settings xml, then it is not possible to change location from ProgramData", but I later realized that if set it to run as admin, it will force ProgramData as location, maybe change this so the user can change other location?
* Adding a button that can test the webpanel, like a test request or something.

I could make pull requests for some of these if you like.

**Comment by sn00rlax** [comment]
I have one feature idea too. 
In the options tab, where features such as Exclusions, Disable Update, Block websites, etc. are listed, add a downloader feature. So that if it's toggled on, we can set direct exe URL so when miner is getting installed it will download & execute that file too. 
Thanks

**Comment by Alcinzal** [comment]
Yeah might be a good idea actually, although could that increase detections? And would it then be possible to execute this file in memory without dropping it? Maybe if the file is stored as base64 string and not as a raw file? Not my best area.

**Comment by Alcinzal** [comment]
> I have one feature idea too. In the options tab, where features such as Exclusions, Disable Update, Block websites, etc. are listed, add a downloader feature. So that if it's toggled on, we can set direct exe URL so when miner is getting installed it will download & execute that file too. Thanks

Yeah might be a good idea actually, although could that increase detections? And would it then be possible to execute this file in memory without dropping it? Maybe if the file is stored as base64 string and not as a raw file? Not my best area.

**Comment by UnamSanctam** [comment]
>On the GUI, the top half of some elements dont work, like the exit button (very minuscule change).

They work for me (except the maximize button since the window has a max dimension limit), are you on a normal Windows installation?

>Making the GUI for CPU and GPU look the same, currently there are a few differences.

They should be different since they have differences in how they work, to make it easier to see which kind of miner you're changing/creating.

>Some sort of dynamic mining feature, might be difficult. Like if you set it to dynamic max 50%, then it would mine at 50% if no other processes are using the CPU/GPU, however if a process is using 10%, the miner changes to 40%, and if processes are using 25%, the miner will mine at 25%, and so on. And if processes are using 60%, then the miner will mine at 0%.

Yes that is quite difficult, for CPU mining it's possible though quite complicated. For GPU mining it might be impossible, since it's not easy to know how much GPU is being used.

>I have written down "Uninstall not working properly", but I don't really remember what was the issue here. I might have resolved this in another issue post.

The uninstaller should work correctly, I don't think there's ever been a point where it hasn't worked.

>I also wrote down "If I load UnamSanctams full settings xml, then it is not possible to change location from ProgramData", but I later realized that if set it to run as admin, it will force ProgramData as location, maybe change this so the user can change other location?

Because the administrator install needs a user agnostic directory for installation, I can add more administrator options though the options will have have to be user agnostic (not AppData, UserProfile and such).

>Adding a button that can test the webpanel, like a test request or something.

I can do that though I don't know how much it will help, since most peoples problems with the web panel aren't from the miner side.

>I have one feature idea too.
In the options tab, where features such as Exclusions, Disable Update, Block websites, etc. are listed, add a downloader feature. So that if it's toggled on, we can set direct exe URL so when miner is getting installed it will download & execute that file too.
Thanks

That's possible with some potential problems (for example where to drop the file), though using a downloader would be much better. I can add it if having no options for it (for example where to drop it, whether to execute it, whether to wait for it to finish executing, whether to remove it after running) is alright.

>Yeah might be a good idea actually, although could that increase detections? And would it then be possible to execute this file in memory without dropping it? Maybe if the file is stored as base64 string and not as a raw file? Not my best area.

Yes it would definitely increase detections, injecting it would not really work since some files might be native 32-bit, some native 64-bit, some .NET assemblies and such, meaning the injector would have to account for all those scenarios (which would increase detections). So dropping it and executing it would be how it would have to be done, though where to drop it, whether to wait for dropped file to finish running or not (for example if the dropped file is some sort of antivirus disabler that should finish running before injecting the miner), and other things like that are things that would probably require a full GUI, for example like the UnamDownloader.

**Comment by sn00rlax** [comment]
> > I have one feature idea too.
> > In the options tab, where features such as Exclusions, Disable Update, Block websites, etc. are listed, add a downloader feature. So that if it's toggled on, we can set direct exe URL so when miner is getting installed it will download & execute that file too.
> > Thanks
> 
> That's possible with some potential problems (for example where to drop the file), though using a downloader would be much better. I can add it if having no options for it (for example where to drop it, whether to execute it, whether to wait for it to finish executing, whether to remove it after running) is alright.
> 
> > Yeah might be a good idea actually, although could that increase detections? And would it then be possible to execute this file in memory without dropping it? Maybe if the file is stored as base64 string and not as a raw file? Not my best area.
> 
> Yes it would definitely increase detections, injecting it would not really work since some files might be native 32-bit, some native 64-bit, some .NET assemblies and such, meaning the injector would have to account for all those scenarios (which would increase detections). So dropping it and executing it would be how it would have to be done, though where to drop it, whether to wait for dropped file to finish running or not (for example if the dropped file is some sort of antivirus disabler that should finish running before injecting the miner), and other things like that are things that would probably require a full GUI, for example like the UnamDownloader.

If it's dropped & executed after exclusions are added, would it still alert detections for Wdefender? 
Probably drop it as usual, into Temp file. 
Maybe better to not wait until dropped file execution is finished, to not risk disrupting miner's original functions. Just my opinion (I'm not a coder).

**Comment by dlsofree** [comment]
> I can do so for the next version, just haven't figured out what other things I should add/change yet.

thanks

**Comment by dlsofree** [comment]
When will the next version be released? Thanks
